### PR TITLE
Fix modification of mutable default value

### PR DESCRIPTION
**Pull request by taus-semmle**
Hi,

I came across what I think is a bug when looking at the alerts for this project on [LGTM.com](https://LGTM.com) (full disclosure: I work on the Python analysis there).

Essentially, default parameter values in Python are shared between all invocations of a function. This means that if a default value is mutable, and if it is modified inside the function, then that modification will be in effect the next time the function is called, and this is rarely what we want. 

In this particular case, the default value for `ignore_fields` will contain all of the `id_fields` encountered so far, and I'm guessing this is not the intended behaviour.

(If I'm mistaken, please do let me know! We're always interested in improving our analysis and eliminating false positives.)

You can see the original alert on LGTM.com here: https://lgtm.com/projects/g/Kinto/kinto/alerts/?mode=tree&ruleFocus=4840097

Cheers,
    Taus

(I will leave these bits below unfilled until you've determined whether this is actually a bug fix or not...)

- [ ] Add documentation.
- [ ] Add tests.
- [ ] Add a changelog entry.
- [ ] Add your name in the contributors file.
- [ ] If you changed the HTTP API, update the [API_VERSION](https://github.com/Kinto/kinto/blob/master/kinto/__init__.py#L15) constant and add an API changelog entry [in the docs](https://github.com/Kinto/kinto/blob/master/docs/api/index.rst)
- [ ] If you added a new configuration setting, update the [`kinto.tpl`](https://github.com/Kinto/kinto/blob/master/kinto/config/kinto.tpl) file with it.

### CODE REVIEW

**Comment by Natim** [comment]
Thank you for taking the time. It looks like the right fix.

**Comment by glasserc** [comment]
Yikes, I didn't think `+=` mutated the original value. Good catch!

**Comment by taus-semmle** [comment]
@Natim Awesome! I'm glad I could be of help.
@glasserc Thank you for the kind words, but really I can't take credit for this -- it was all down to our automated analysis. Subtle bugs of this kind can be very difficult to catch "by hand".
@leplatrem I'll add a change note. I also just realised that the `def validate_from_bucket_schema_or_400` function has the same problem, so I'll add a fix for that. (I also notice that our analysis _didn't_ catch that instance, so I'll be looking into extending it.)

Incidentally, I notice that the tests are failing due to lack of coverage. I'm not terribly familiar with Travis CI, but I guess this means the function is never actually called in such a way that the default parameter is used?

**Comment by Natim** [comment]
> I notice that the tests are failing due to lack of coverage.  I guess this means the function is never actually called in such a way that the default parameter is used.

The default value is not used by our tests. Maybe we should add a test that uses it?

**Comment by Natim** [comment]
I checked and validate_schema is used in only two places were ignore_fields is always defined.

**Comment by leplatrem** [comment]
Can you please update CHANGELOG + CONTRIBUTORS files?

**Comment by taus-semmle** [comment]
I'm doing it now. Sorry for the delay!

File: CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,7 @@ This document describes changes between each past release.
 
 **Bug fixes**
 
+- Fixed two potential bugs relating to mutable default values.
```
[the remark this answers is not in the document]
Comment (taus-semmle) [reply CHANGELOG.rst]: :man_facepalming: Thanks for pointing this out!

**Pull request merged**